ample/ensembler/truncation_util.py

```
@@ -32,204 +39,98 @@ def calculate_residues_focussed(var_by_res):
-    
     # Calculate the percentages for each of these start points
     percentages = [ int(round(float(start + 1) / float(length) * 100)) for start in start_indexes ]
     # print "percentages ", percentages
     truncation_levels = percentages
-
-    # print "var_by_res ",var_by_res
     idxs_all = [ x.idx for x in var_by_res ]
     resseq_all = [ x.resSeq for x in var_by_res ]
     variances = [ x.variance for x in var_by_res ]
-
     truncation_residue_idxs = [ sorted(idxs_all[:i + 1]) for i in start_indexes ]
-    # print "truncation_residue_idxs ",truncation_residue_idxs
     truncation_residues = [ sorted(resseq_all[:i + 1]) for i in start_indexes ]
-    # print "truncation_residues ",truncation_residues
-    
     # We take the variance of the most variable residue
     truncation_variances = [ variances[i] for i in start_indexes ] 
-    # print "truncation_variances ",truncation_variances
     
     return truncation_levels, truncation_variances, truncation_residues, truncation_residue_idxs
 
 
-def calculate_residues_percent(var_by_res, percent_interval):
-    """Calculate the list of residues to keep if we are keeping self.percent residues under
-    each truncation bin. The threshold is just the threshold of the most variable residue"""
-    
-    MIN_CHUNK = 3  # We need at least 3 residues for theseus to work
-    length = len(var_by_res)
-    start_idxs = _split_sequence(length, percent_interval, min_chunk=MIN_CHUNK)
+def _calculate_start_indexes_from_fixed_percentages(percent_fixed_intervals, length, all_indexes):
+    "Calculate where in the list of residues each percentage bin starts"
+    def find_closest_index(nresidues_ideal, all_indexes):
+        min_dist = 99999 
+        start = None
+        for idx in all_indexes:
+            rdist = abs(nresidues_ideal - (idx + 1))
+            if  rdist <= min_dist and idx + 1 >= MIN_CHUNK:
+                min_dist = rdist
+                start = idx
+        assert start is not None
```

Better to raise an Exception than the assert here. It won't make any sense when receiving log files containing this error.

---

It's just a failsafe that should never be triggered